loopback.cc

```diff
@@ -35,10 +34,18 @@ int loopback_init(const char* device, int w, int h, int debug) {
 	int ret_code = 0;
 
 	fdwr = open(device, O_RDWR);
-	assert(fdwr >= 0);
+	if(fdwr < 0) {
+		perror(__func__);
+		return -1;
+	}
 
 	ret_code = ioctl(fdwr, VIDIOC_QUERYCAP, &vid_caps);
-	assert(ret_code != -1);
+	if(ret_code < 0) {
+		perror(__func__);
+		printf("The device capabilities could not be queried.\n");
+		close(fdwr);
+		return -1;
+	}
 
```

Comment:
As above, consistent output to `stderr` and line number, or unique descriptive message for identification.